src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -239,7 +240,9 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
                 {
                     // Create the comment file for reshelve.
                     StringBuilder comment = new StringBuilder(tfShelveset.Comment ?? string.Empty);
-                    if (!(executionContext.Variables.Build_GatedRunCI ?? true))
+                    string runCi = GetEndpointData(endpoint, Constants.EndpointData.GatedRunCI);
+                    bool? gatedRunCi = executionContext.Variables.GetBoolean(runCi);
```

Comment:
I am not getting it from variable dictionary. GetBoolean is a helper method that parses a string and returns a nullable bool. I am just reusing the public method . Hope thats fine :)